core/trino-main/src/main/java/io/trino/execution/TaskId.java

```diff
@@ -35,56 +33,61 @@
     @JsonCreator
     public static TaskId valueOf(String taskId)
     {
-        return new TaskId(taskId);
+        requireNonNull(taskId);
+        var parts = parseDottedId(taskId, 4, "taskId");
+        StageId stageId = StageId.valueOf(parts.subList(0, 2));
+        int partitionId = parseInt(parts.get(2));
+        int attemptId = parseInt(parts.get(3));
+        return new TaskId(stageId, partitionId, attemptId);
     }
 
-    private final String fullId;
+    private final StageId stageId;
+
+    private final int partitionId;
+
+    private final int attemptId;
 
     public TaskId(StageId stageId, int partitionId, int attemptId)
```

Comment:
I'd use a builder or static factory as it is easy to change in a backwards compatibly in the future.

Reply:
Same comment for StageId

Reply:
If we go with https://github.com/trinodb/trino/pull/14445#discussion_r987308437 I think we do not need TaskId/StageId in SPI at all.

Reply:
I have removed all the changes to TaskId and StageId, and instead added a simple `ConnectorPageSinkId` in trino-spi, as we discussed.